l2geth/rollup/sync_service.go

```diff
@@ -303,371 +305,308 @@ func (s *SyncService) VerifierLoop() {
 	}
 }
 
+// verify is the main logic for the Verifier. The verifier logic is different
+// depending on the SyncType.
 func (s *SyncService) verify() error {
-	// The verifier polls for ctc transactions.
-	// the ctc transactions are extending the chain.
-	latest, err := s.client.GetLatestTransaction()
-	if err != nil {
-		return err
-	}
-
-	if latest == nil {
-		log.Debug("latest transaction not found")
-		return nil
-	}
-
-	var start uint64
-	if s.GetLatestIndex() == nil {
-		start = 0
-	} else {
-		start = *s.GetLatestIndex() + 1
-	}
-	end := *latest.GetMeta().Index
-	log.Info("Polling transactions", "start", start, "end", end)
-	for i := start; i <= end; i++ {
-		tx, err := s.client.GetTransaction(i)
+	switch s.syncType {
+	case SyncTypeBatched:
+		err := s.syncTransactionBatchesToTip()
 		if err != nil {
-			return fmt.Errorf("cannot get tx in loop: %w", err)
+			log.Error("Verifier cannot sync transaction batches to tip", "msg", err)
 		}
-
-		log.Debug("Applying transaction", "index", i)
-		err = s.maybeApplyTransaction(tx)
+	case SyncTypeSequenced:
+		err := s.syncTransactionsToTip(SyncTypeSequenced)
 		if err != nil {
-			return fmt.Errorf("could not apply transaction: %w", err)
+			log.Error("Verifier cannot sync transactions with SyncTypeSequencer", "msg", err)
 		}
-		s.SetLatestIndex(&i)
 	}
```

Comment:
This switch logic seems error prone to me? 

Seems better for us to have a `ReplicaSyncService` and a `L1SyncService` and have them internally implement the required logic for each of the functionalities explicitly.

Reply:
The `SyncService.syncType` is not dynamic and is configured up front. You are worried about misconfiguration? 

We would need `SequencerSyncService`, `ReplicaSyncService` and `L1SyncService` - they share a lot of functionality and breaking them out would add a few hundred lines of code. I'd definitely prefer to not break them apart